### PR TITLE
Disable gloo by default

### PR DESCRIPTION
**Pull request by sandyhouse**
### PR types

Bug fixes
### PR changes

Others
### Describe

fix the bug in gloo initialization for pserver by only initializing gloo for worker and disable gloo initialization by default.
Users can also enable gloo using `export PADDLE_WITH_GLOO=1`

### CODE REVIEW

File: python/paddle/distributed/fleet/base/role_maker.py [hunk 1]
```diff
@@ -220,15 +220,8 @@ def init(rank, nodes, role):
             rank, nodes = self._get_rank_nodes(Role.WORKER)
             gloo = init(rank, nodes, "WORKER")
             self._worker_comm = gloo
-        else:
-            rank, nodes = self._get_rank_nodes(Role.SERVER)
-            gloo = init(rank, nodes, "SERVER")
-            self._server_comm = gloo
```
Comment (danleifeng) [inline python/paddle/distributed/fleet/base/role_maker.py:226]: should request @seiriosPlus for review.

File: python/paddle/distributed/parallel.py [hunk 2]
```diff
@@ -142,21 +142,23 @@ def _check_var_exists(var_name):
...
-    if parallel_env.rank == 0:
-        # The scope for worker used by http server is '_worker'
-        size = {'_worker': parallel_env.world_size}
-        http_server = Process(
-            target=_start_kv_server,
-            args=(int(ep_rank_0[1]), http_server_d, size))
-        http_server.daemon = True
-        http_server_d["running"] = True
-        http_server.start()
+    init_gloo = int(os.getenv("PADDLE_WITH_GLOO", "0"))
```
Comment (danleifeng) [inline python/paddle/distributed/parallel.py:145]: 使用flags还是一个接口的参数去控制gloo的开启，这里是怎么计划的呢

Comment (sandyhouse) [reply python/paddle/distributed/parallel.py:145]: 目前是通过PADDLE_WITH_GLOO环境变量控制的。

**Review by chalsliu — APPROVED** [review]

**Review by seiriosPlus — APPROVED** [review]
LGTM

**Pull request merged**